### PR TITLE
test: rbd master/slave notify test should test active features

### PR DESCRIPTION
**Pull request by dillaman**
http://tracker.ceph.com/issues/19692

### CODE REVIEW

File: src/test/librbd/test_notify.py [hunk 1]
```diff
@@ -141,8 +141,10 @@ def slave(ioctx):
         assert(list(image.list_snaps()) == [])
 
         print("rebuild object map")
-        image.update_features(RBD_FEATURE_OBJECT_MAP | RBD_FEATURE_FAST_DIFF,
-                              False)
+        features = image.features() & (
+                RBD_FEATURE_OBJECT_MAP | RBD_FEATURE_FAST_DIFF)
+        if features:
+            image.update_features(features, False)
         image.update_features(RBD_FEATURE_OBJECT_MAP, True)
```
Comment (smithfarm) [inline src/test/librbd/test_notify.py]: It's only disabling the feature that causes the exception, eh? Enabling it (as in this line) is not a problem?

Comment (dillaman) [inline src/test/librbd/test_notify.py]: Jewel throws an "InvalidArgument" exception if you try to disable a feature that wasn't enabled (or try to enable a feature that was already enabled).

**Comment by smithfarm** [comment]
I prematurely cherry-picked this to kraken at #14641 and am running tests on it.

**Comment by smithfarm** [comment]
@dillaman Oddly enough this does not fix the issue in kraken: http://pulpito.ceph.com/smithfarm-2017-04-20_08:23:44-upgrade:client-upgrade-wip-kraken-backports-distro-basic-vps/

The sha1 being tested is 0277bbc (wip-kraken-backports) and I double-checked that it does include #14641.

**Comment by dillaman** [comment]
@smithfarm That one test pulls the test specifically from the "jewel" branch -- so I wouldn't expect it to work until it's fully backported (or we could have a temporary tweaked version of the suite test that pulls from a wip-jewel-backports branch)

**Comment by smithfarm** [comment]
@dillaman This may be impulsive, but my vote would be to merge the master, kraken, and jewel backports without further ado, unless you can think of an easier way to test them.

**Comment by dillaman** [comment]
@smithfarm Since it's just test cases, I would agree.

**Pull request merged**